### DISCUSSION TITLE
v3.0.0 Release

### DISCUSSION THREAD

**Discussion by pranavrajs** [chatwoot/chatwoot#7570] [Releases & Announcements]
Here is an update on the v3 release. Our theme for the release was helping with prioritization and improving efficiency. Here are the main changes we are introducing. 

### 1/ A Native AI Assistant: 

This is an intelligent assistant that can help you throughout the entire customer support lifecycle. Here are some of the things it can do:

- **Talk to customers:** It can immediately respond to customers, solving most generic queries without needing your interaction.

- **Triage conversations:** It helps you prioritize conversations better by automatically generating labels and assisting with replies.

- **Tailor content:** The AI assistant learns from your interactions with customers and generates articles that were not available on your website. This helps you identify missing pieces and add more contextual content.

We believe that this would reduce the conversations (solves first-level interactions easily) dramatically and would help you prioritize the conversation.

### 2/ Improved messenger:

We have completely redesigned the website widget, making it faster and adding support for the help center. The redesign includes additional branding elements to prioritize your brand’s perception. This update also addresses longstanding issues, such as continuity issues and support for multiple thread conversations.

### 3/ Improvements in reporting:

We have introduced a new metric that enables you to measure the duration of time a customer waited for a response from an agent. We consider this metric, in conjunction with the first response time, to be crucial for monitoring customer service performance. In future releases, we plan to include automation and SLA features to further enhance this metric's usefulness.

### 4/ Support for dark mode on the dashboard:

There has been a lot of discussion within the community regarding dark mode. We took this opportunity to clean up the old codebase, and thanks to TailwindCSS, we were able to get this up and running in a short amount of time. We are in the process of fully migrating from FoundationCSS.

### 5/ Sentiment Analysis on conversations (Experimental): 

We are also working on analyzing the conversation to see if it can be used to prioritize the conversation. 

### 6/ Fixes for commonly reported issues

We are addressing some of the outstanding community issues, such as permissions, email channel stability, and confusion around the WhatsApp token authentication flow.

### The release process
The obvious question is: how do you get these updates on your instances? Let's delve into that.

As you may have seen in some of the commits, we have many dependency updates, including the Postgres pg_vector extension. Although upgrading to version 15 should work without issues, we have not tested it thoroughly on all platforms, and some planned updates are still awaiting quality checks.

Due to these concerns, we will not follow the usual monthly release cycle due to the aforementioned concern. Today, we will release the first candidate with the changes mentioned. During the next week, we will continue with the testing and follow-up on the stable releases.

The changes will not be available in the master branch immediately. To get the changes to your instance, you will have to follow the [manual upgrade process](https://www.chatwoot.com/docs/self-hosted/deployment/linux-vm#upgrading-to-a-newer-version-of-chatwoot) with the branch name 3.0.0-rc1. Please note that we do not have the documentation for upgrading instances or fixing common issues. Therefore, I would advise against switching your production instances immediately.

Let us know if you have any questions.

**Comment by LiamAshdown** [comment]
This is great!

Is there a demo which we can see the V3?

**Reply by jtayrone** [reply]
You need to have a chatwoot instance and make it switch from white to 3.0.0-rc1.

If not, check the [manual update process](https://www.chatwoot.com/docs/self-hosted/deployment/linux-vm/#upgrading-to-a-newer-version-of-chatwoot).

**Reply by pranavrajs** [reply]
As @jtayrone suggested, this is a manual process. You can follow the guide linked in the above thread and use the branch `develop` instead of master to get the changes. This is only a step until the time being. You would be able to upgrade it in the usual upgrade format once a stable release is created.

**Reply by noezdev** [reply]
He just asked for a demo not for an upgrade manual...

**Comment by DanielK1511** [comment]
Sorry English isn't my first Language, is there already a way to upgrade to 3.0.0? I Can't seem to figure it out using the linked upgrade process.

Cheers

**Reply by guiqis** [reply]
They said you need to do a Manual Upgrade to get v3

**Reply by pranavrajs** [reply]
Please see the comment above.

**Comment by TalisonSantana** [comment]
about the permissions, what changes are being made specifically?

**Reply by pranavrajs** [reply]
@TalisonSantana You should be able to view the conversations just by adding the team members to team rather than adding them to the inboxes.

**Reply by kleberit** [reply]
Hello @pranavrajs these permissions are already in this first version 3.0, or will be included in future releases, I did tests and saw no difference in the behavior of version 2.18 to version 3.0

**Reply by ammaraslam** [reply]
@pranavrajs @sojan-official about teams permissions, can we add teams in inbox collaborator as well instead of adding individual agent so that chats are assigned round robin wise to that team only?

Having both options would work too, being able to add agents/teams as collaborators.

Is your approach gonna be the same?

**Reply by ClaudioNoronha** [reply]
@pranavrajs Viewing messages only by teams without the need to include agents in inboxes is not yet available right? I installed version 3.0 here and still haven't noticed any difference compared to version 2.18

**Reply by Shiba1805** [reply]
that's not available yet, right??

**Comment by jtayrone** [comment]
Grateful! It is gratifying to follow the evolution of a tool with an engaged community and dev.

Congratulations on everything 👍😎🚀

**Reply by pranavrajs** [reply]
Thanks :)

**Comment by converso-chat** [comment]
Good morning everybody!
I did the update, but it didn't reflect any visual changes on our dash
board. And it keeps showing v2.18

Em sex., 21 de jul. de 2023 às 20:08, Tayrone Santos <
***@***.***> escreveu:

**Reply by jtayrone** [reply]
You need to switch to branch name 3.0.0-rc1

**Comment by converso-chat** [comment]
Hello
Tayrone Santos, where do I change the branch? I used this guide
https://www.chatwoot.com/docs/self-hosted/deployment/linux-vm/#upgrading-to-a-newer-version-of-chatwoot
to upgrade. as I always do

Thanks

Em sáb., 22 de jul. de 2023 às 09:23, Tayrone Santos <
***@***.***> escreveu:

**Reply by pranavrajs** [reply]
You can use the latest `develop` or `v3.0.0-rc1`.

https://github.com/chatwoot/chatwoot/tree/v3.0.0-rc1

**Comment by Michaelmallmann** [comment]
Hi! I do have one question! As you mentioned … the AI will answer customers about generic topics, and it will learn from the interactions we have on Chatwoot. So far so good. 

My question: Did you consider adding a field (or some other solution) in which we would add knowledge to the AI? 

OpenAI just added two new fields to Settings so we can “teach” ChatGPT a few parameters. Maybe it would be a good ideia.

**Reply by pranavrajs** [reply]
@Michaelmallmann If you are talking about more articles for it to learn, we have considered that. You should be able to customize the content it learns from. If it is something else, please let me know more details.

**Reply by Michaelmallmann** [reply]
@pranavrajs that’s exactly it! Thank you.

**Comment by converso-chat** [comment]
Hello! I updated to v3.0 with develop branch. But I didn't find any changes

Em ter., 25 de jul. de 2023 às 06:27, noezdev ***@***.***>
escreveu:

**Reply by cruz4d3r** [reply]
I have this issue too. @pranavrajs

**Reply by uhlhosting** [reply]
I believe this is ongoign. And by now you should upgrade to the stable 3 release. Do not use other branches unless when told so by chatwoot team.

**Comment by pucilpet** [comment]
Just so you know, there's no v3.0.0-rc1 image on docker hub, only v3.0.0-rc1-ce.

I upgraded to develop image but could not see how this was implemented: "We have completely redesigned the website widget... adding support for the help center."

Is there something we need to do in addition to choosing the help desk in the channel settings?

**Comment by MarconiD7** [comment]
Toda vez que o mesmo cliente chama, o chat é reaberto com o mesmo número de protocolo e ainda com o nome do agente anterior. Assim causa um problema de controle de quantidade de chamados no chat, pois se o cliente chama varias vezes ele sempre tera o mesmo protocolo e também fica dificil para conseguir controlar quem é o responsavel pelo chamado pois sempre fica para o ultimo responsavel quando o cliente abre o chamado, dificultando verificar o historico de conversa.
Poderia que cada vez que tivesse uma nova conversa, gera - se um novo protocolo e não ao mesmo prontocolo da conversa anterior. Assim cada conversa teria o seu protocolo facilitando de verificar o historico.

Isso foi corrigido?

**Reply by luizeof** [reply]
qual canal que vc está usando ?

**Reply by MarconiD7** [reply]
API 360Dialog

**Reply by glauberssoutodossantos** [reply]
@MarconiD7 existe uma funcionalidade de bloqueio para conversa única, mas isto apenas funciona para conversas ativas. As receptivas vão sempre cair na mesma conversa da anterior.

Minha sugestão para resolver teu problema seria criar um bot, que sempre que a conversa é reaberta, gerar um novo número de protocolo aleatório (diferente do id da conversa que o Chatwoot te dá). Esse mesmo bot também pode atribuir aleatoriamente a uma pessoa ou a um time.

**Reply by MarconiD7** [reply]
@glauberssoutodossantos Só não entendi ainda essa funcionalidade da opção estar ativada ou desabilitada, eu testei as duas e não identifiquei nenhuma mudança.  
Sobre a questão do Bot, seria criado na Dialogflow? é que atualmente utilizamos pelo "Automação" do próprio Chatwoot.

**Reply by glauberssoutodossantos** [reply]
Sobre a funcionalidade de conversa única, basicamente muda o comportamento desse botão.

![image](https://github.com/chatwoot/chatwoot/assets/99494469/e1c2d6eb-bf61-46ed-8f64-576484e2dbb5)

Se estiver ativo: Quando enviado uma mensagem PROATIVA, NÃO GERA, uma nova conversa.
Se estiver desativo: Quando enviado uma mensagem PROATIVA, GERA-SE, uma nova conversa.

Sobre o bot eu recomendo a utilização do dialogflow.

**Reply by MarconiD7** [reply]
@glauberssoutodossantos Perfeito, agora eu entendi. Muito obrigado pela explicação, abraço!

**Comment by jtayrone** [comment]
Let's go:
First which api channel are you using? This treatment comes from the api not from chatwoot, in the Whatsapp Cloud API Provider for example you have the option to enable multiple conversations in your input box.

![lok-single](https://github.com/chatwoot/chatwoot/assets/34479816/277549a6-538f-4957-9d18-b77dd517552d)

**Reply by MarconiD7** [reply]
Utilizo a API 360Dialog do WhatsApp e essa opção está como Disabled. Entrei em contato com o suporte da 360Dialog e informaram que da parte deles está tudo ok e que seria algum problema no chatwoot.

**Reply by glauberssoutodossantos** [reply]
@MarconiD7 essa funcionalidade é apenas para conversas ativas. Para conversas receptivas o chatwoot vai sempre tentar procurar a última conversa salva.

**Comment by brendewguedes** [comment]
When we create a tag and put it on the client, it doesn't show on contacts. This is a mistake, since we already add tags during a conversation, and when we go to contacts we must put them again. This delays, moreover at the time of the report it will get in the way. the ideal would be that when we place the labels during a conversation, it is used in the contacts' labels automatically.

**Comment by sanjayws** [comment]
Upgraded this on my dev environment and looks cool. 

Question, is dark mode available? If so, how do we turn it on?

Thank you, again, for amazing work around CW!

**Comment by sanjayws** [comment]
Was just testing out the super admin from this v3 pre-release, when updating account, e.g. selecting any options and saving it, it throws an error. Either editing an existing or creating a new one throws errors.

"We're sorry, but something went wrong.
If you are the application owner check the logs for more information."

Here are the corresponding logs:
```
Aug  2 17:35:13 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:13.971152 #120163]  INFO -- : [0665ca2b-b07f-4e78-8943-4e0d8b34389c] Started GET "/super_admin/accounts/2/edit" for 172.69.85.140 at 2023-08-02 17:35:13 +0800
Aug  2 17:35:13 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:13.973211 #120163]  INFO -- : [0665ca2b-b07f-4e78-8943-4e0d8b34389c] Processing by SuperAdmin::AccountsController#edit as HTML
Aug  2 17:35:13 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:13.973260 #120163]  INFO -- : [0665ca2b-b07f-4e78-8943-4e0d8b34389c]   Parameters: {"id"=>"2"}
Aug  2 17:35:13 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:13.987017 #120163]  INFO -- : [0665ca2b-b07f-4e78-8943-4e0d8b34389c]   Rendered /usr/local/rvm/gems/ruby-3.2.2/gems/administrate-0.19.0/app/views/administrate/application/edit.html.erb within layouts/super_admin/application (Duration: 8.6ms | Allocations: 7862)
Aug  2 17:35:13 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:13.993884 #120163]  INFO -- : [0665ca2b-b07f-4e78-8943-4e0d8b34389c]   Rendered layout layouts/super_admin/application.html.erb (Duration: 15.5ms | Allocations: 13370)
Aug  2 17:35:13 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:13.994257 #120163]  INFO -- : [0665ca2b-b07f-4e78-8943-4e0d8b34389c] Completed 200 OK in 21ms (Views: 16.0ms | ActiveRecord: 0.6ms | Allocations: 15335)
Aug  2 17:35:13 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:13.995233 #120163]  INFO -- : source=rack-timeout id=0665ca2b-b07f-4e78-8943-4e0d8b34389c timeout=15000ms service=25ms state=completed
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:15.750144 #120163]  INFO -- : source=rack-timeout id=0dcc4de7-18cd-42b3-88dc-c2ebf164224d timeout=15000ms state=ready
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:15.750673 #120163]  INFO -- : [0dcc4de7-18cd-42b3-88dc-c2ebf164224d] Started PATCH "/super_admin/accounts/2" for 172.69.85.143 at 2023-08-02 17:35:15 +0800
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:15.752863 #120163]  INFO -- : [0dcc4de7-18cd-42b3-88dc-c2ebf164224d] Processing by SuperAdmin::AccountsController#update as HTML
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:15.753030 #120163]  INFO -- : [0dcc4de7-18cd-42b3-88dc-c2ebf164224d]   Parameters: {"authenticity_token"=>"[FILTERED]", "account"=>{"name"=>"ORENMY", "locale"=>"en", "status"=>"active", "limits"=>{"agents"=>"100", "inboxes"=>"100"}}, "enabled_features"=>{"feature_inbound_emails"=>"true", "feature_channel_email"=>"true", "feature_channel_facebook"=>"true", "feature_channel_twitter"=>"true", "feature_ip_lookup"=>"true", "feature_email_continuity_on_api_channel"=>"true", "feature_help_center"=>"true", "feature_agent_bots"=>"true", "feature_macros"=>"true", "feature_agent_management"=>"true", "feature_team_management"=>"true", "feature_inbox_management"=>"true", "feature_labels"=>"true", "feature_custom_attributes"=>"true", "feature_automations"=>"true", "feature_canned_responses"=>"true", "feature_integrations"=>"true", "feature_voice_recorder"=>"true", "feature_mobile_v2"=>"true", "feature_channel_website"=>"true", "feature_campaigns"=>"true", "feature_reports"=>"true", "feature_crm"=>"true", "feature_auto_resolve_conversations"=>"true"}, "commit"=>"Update Account", "id"=>"2"}
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:15.757144 #120163]  INFO -- : [0dcc4de7-18cd-42b3-88dc-c2ebf164224d] Completed 500 Internal Server Error in 4ms (ActiveRecord: 0.4ms | Allocations: 1831)
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: F, [2023-08-02T17:35:15.757614 #120163] FATAL -- : [0dcc4de7-18cd-42b3-88dc-c2ebf164224d]
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: [0dcc4de7-18cd-42b3-88dc-c2ebf164224d] ArgumentError (wrong number of arguments (given 1, expected 0)):
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: [0dcc4de7-18cd-42b3-88dc-c2ebf164224d]
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: [0dcc4de7-18cd-42b3-88dc-c2ebf164224d] app/dashboards/account_dashboard.rb:89:in `permitted_attributes'
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: [0dcc4de7-18cd-42b3-88dc-c2ebf164224d] app/controllers/super_admin/accounts_controller.rb:37:in `resource_params'
Aug  2 17:35:15 play1 chatwoot-web.1[120163]: I, [2023-08-02T17:35:15.757792 #120163]  INFO -- : source=rack-timeout id=0dcc4de7-18cd-42b3-88dc-c2ebf164224d timeout=15000ms service=8ms state=completed
```

**Reply by LiamAshdown** [reply]
This has recently been fixed: #7644 

You need to pull in the develop branch if you want the latest fixes as I don't believe the v3 branch has not pulled in the latest fixes yet.

**Reply by sanjayws** [reply]
OK noted.

**Comment by KonsoleLabs** [comment]
thank you for the first rc. 
I have discovered how to use OpenAI to help create answers, but have not yet managed to get the AI to give answers consistently. Or did I misunderstand part of your announcement? 
All in all, I'm looking forward to the new version ... I'll be happy when Mastodon is added as an inbox. 
Thank you !

**Reply by uhlhosting** [reply]
How do you enable the open ai to automatically reply? Havent found that setting yet.

**Reply by KonsoleLabs** [reply]
YOUR_URL/app/accounts/1/settings/applications

**Reply by uhlhosting** [reply]
there is enabled, yet this only allows me to use the manual button, I dont see how the replies are automated.

**Reply by KonsoleLabs** [reply]
sorry, miss-read your post. I am looking, too

**Comment by sadden3194** [comment]
Hi all, I have an issue after update 3.0.0.

If I go to the settings of my inboxes and click on "Update" I get an Internal server error.
Browser console:

```
XHRPATCH
https://mydomain.de/api/v1/accounts/1/inboxes/1
[HTTP/2 500 Internal Server Error 166ms]
```


@pranavrajs @sojan-official

**Reply by pranavrajs** [reply]
@Ghost108 Please share the error log from the server. Without that, it is hard to find out the root cause.

**Reply by sadden3194** [reply]
@pranavrajs sorry, but I cant ... now it works without any errors :D 
But another topic: shouldn't the web widget be optimized with version 3? see no difference ...

**Reply by sojan-official** [reply]
@Ghost108  The widget changes are still being worked on, We'll be rolling out these features gradually.

**Reply by jtayrone** [reply]
From what I understand, your error is related to your server. I simulated your issue here, and the "problem" was resolved simply by recompiling Chatwoot; the error disappeared. Another factor to consider is the quality of the network you're using. There might be a significant timeout between you and your server.

Also, check the logs to see what they show:

`cwctl --weblogs`

I hope this helps you.

**Reply by nestordavalos** [reply]
turn off response_bot, in super_admin to the accounts you want to create a new inbox, and you'll be fine, this only happens with the channel API

**Comment by eddietoth** [comment]
Recently installed V3 on my instance, and I'm loving the new look! The dark mode is sleek, the fonts have been updated, and the overall experience is much better. I'm excited about what V3 has in store for features, and not just the looks too. 👍

**Comment by sadden3194** [comment]
> @Ghost108 The widget changes are still being worked on, We'll be rolling out these features gradually.
any idea how long will it take?

would like to draw attention to this proposal once again:
https://github.com/chatwoot/chatwoot/issues/5962#issuecomment-1330654172

In addition, it would also be good that text input is not possible for the user if a bot has previously sent a message with "reply buttons".

**Comment by Shivam8584** [comment]
How to activate dark mode on dashboard. Cant see a option

**Reply by sadden3194** [reply]
left bottom corner > click on you profile > change appearance

**Reply by iamsivin** [reply]
@Shivam8584 You can access it in two ways.
1. By command bar. Press `cmd` + `k` and search for Change Appearance.
2. Or from the profile menu that @Ghost108 has shared.

**Reply by Shivam8584** [reply]
![16923432596755403654777058855962](https://github.com/chatwoot/chatwoot/assets/120591342/895b5c08-2510-4a62-bab1-7da79d42c402)

Nothing coming up here. Are there any additional steps after cwctl --upgrade ? Version 3.0.0 shows up in settings but I am not seeing any changes

**Reply by Shivam8584** [reply]
I was on version 2.18

**Reply by iamsivin** [reply]
@Shivam8584 The dark mode feature will be available from v3.0.

**Reply by ShivamGupta-SM** [reply]
already on 3.0. but still not visible

**Reply by cruz4d3r** [reply]
I have this issue

**Comment by Shivam8584** [comment]
What are next steps after ctcwl --upgrade
I have updated by cant see any changes except from version number

**Comment by carleii** [comment]
I love this release. Thanks for all the contributors

**Comment by Philipinho** [comment]
For anyone using Caprover.
You can upgrade using the following Caprover captain-definition file (Method 5).
Paste to the "Method 5: Deploy captain-definition file" and click on "Deploy".

### Caprover Web service
```
{
  "schemaVersion": 2,
  "dockerfileLines": [
    "FROM chatwoot/chatwoot:v3.0.0",
    "RUN chmod +x docker/entrypoints/rails.sh",
    "ENTRYPOINT [\"docker/entrypoints/rails.sh\"]",
    "CMD bundle exec rake db:chatwoot_prepare; bundle exec rails s -b 0.0.0.0 -p 3000"
  ]
}
```

### Caprover Worker service
```
{
  "schemaVersion": 2,
  "dockerfileLines": [
    "FROM chatwoot/chatwoot:v3.0.0",
    "RUN chmod +x docker/entrypoints/rails.sh",
    "ENTRYPOINT [\"docker/entrypoints/rails.sh\"]",
    "CMD bundle exec sidekiq -C config/sidekiq.yml"
  ]
}
```

Make sure to backup your postgres database before upgrades.
On the linux server side, you can use this command to backup your db.

```
docker exec CHATWOOT_POSTGRES_CONTAINER_NAME_HERE pg_dump chatwoot -U chatwoot -F c > chatwoot_backup-21-02-23.dump
```

pg_dump chatwoot -> assumes chatwoot is the db name. Replace if not.
-U chatwoot -> assumes chatwoot is the db user. Replace if not.

CHATWOOT_POSTGRES_CONTAINER_NAME_HERE -> use the name of the chatwoot running container.

**Comment by AdsonCicilioti** [comment]
Hi guys!

Are messages with `cards` or interactive messages, yet working on the Dashboard?

#609 (#594) Isn't clear for me. The title says "...Widget and Dashboard".
Again... This is working on the Dashboard (Inboxes conversations)?

**Comment by HermesMacedo** [comment]
@pranavrajs #6456 The ticks like WhatsApp for send, received, unread and read message, to help users verify the status of messages sent, via API.
Hi, any update on this? Has this option already been implemented? at least for whatsapp?

**Reply by nestordavalos** [reply]
Hello, how are you, I hope there is a pull that may help, check it out
https://github.com/chatwoot/chatwoot/pull/6038

**Comment by jordan26** [comment]
Is v3 stable now?

**Reply by uhlhosting** [reply]
@jordan26 v3 worked fine for all those who had clean install, in some particular cases some people had issues migrating, yet most of times from what I have notice in these conversations, some people have more and some less knowledge of the tasks they do, based on that many issues can occur and complains, some people are lazy to read the docs, yet are happy to shout here about issue... Yet I work with v3 since its release and had no major incidents.

**Comment by Evoltstore** [comment]
Sad about no VOIP :(
